docs/auth/index.md

```diff
@@ -10,6 +10,8 @@ configure Backstage to have any number of authentication providers, but only
 one of these will typically be used for sign-in, with the rest being used to provide
 access external resources.
 
+> Note: Backstage backend APIs are by default unauthenticated. Thus, if your Backstage instance is exposed to the Internet, anyone can access information in the Backstage. If you would like to learn more, read about how to [authenticate API requests"](https://github.com/backstage/backstage/blob/master/contrib/docs/tutorials/authenticate-api-requests.md).
```

Comment:
Certainly, but I would like to get more clarification on these 3 texts.
1. Warning in https://backstage.io/docs/overview/threat-model/#integrator-responsibilities
2. Note in https://backstage.io/docs/auth/#sign-in-configuration
3. This PR note

My main goal with the suggested note is to make it explicit that the API exposed, and I would not consider the first and second texts to be related to my suggestion. However, if we want to make it clear that the API is exposed and the main idea is for the "Integrator" to ensure security, then I think I should actually edit the second note no? 

Would the following be more preferred? Include an explicit warning that the API is unauthenticated in the second note, with also a link to the first warning, instead of my additional note.

Also, this might not be the best place to discuss this, as I don't know the whole architectural decisions of Backstage, but are you sure not including link to the cookie tutorial is desired? I am somewhat confused as to why Backstage provides so many features for Identity management, authentication, and secure communication between plugins, but doesn't provide "the whole package". If you already strongly suggest to run backstage behind an Oauth proxy or some auth layer, wouldn't it be easier for backstage devs to drop these features, and shift all work towards the "owner"? Authorization included.

Reply:
I'm thinking we get rid of the second note and put it all in this top one instead.

We want to provide full auth out of the box, but it's WIP. We don't have the bandwidth to pick it up yet but we're hoping to have it done this year. Bit more info over here: #15999

Reply:
@VladMasarik, will you be able to make the above change, please? Once that's done we can get this approved and merged in

Reply:
@awanlin @Rugvip sorry for the hold up, I have merged the two texts, and added a link to the suggested "threat model". I am not sure it is the best, so I am very open to any suggestions.